R/post_dataset_utils.R

```diff
@@ -0,0 +1,328 @@
+#' Create a json query
+#'
+#' @description
+#' Creates a json query for use when POST-ing a query to the API. This takes time period,
+#' geography, filter item and indicator criteria and produces a working json query as a
+#' single string. The result can be used directly by post_dataset() or the output of
+#' `parse_tojson_params(...) |> cat()` can be copied and pasted as the "body" content in
+#' other API connection software (such as Postman) to POST a query to the EES API.
+#'
+#' @inheritParams api_url
+#' @inheritParams parse_tojson_geographies
+#'
+#' @return String containing json query body for use with http POST request
+#' @export
+#'
+#' @examples
+#' parse_tojson_params(example_id("indicator")) |>
+#'   cat()
+#'
+#' parse_tojson_params(
+#'   example_id("indicator"),
+#'   time_periods = "2024|W23",
+#'   geographies = c("NAT|id|dP0Zw", "REG|id|rg3Nj"),
+#'   filter_items = c("pmRSo", "7SdXo")
+#' ) |>
+#'   cat()
+#'
+#' # Create a geographies data frame to find both of:
+#' #   - England national level data
+#' #   - all LAs in a specified region ("E12000004")
+#' dfgeographies <- data.frame(
+#'   return_level = c("NAT", "LA"),
+#'   search_level = c("NAT", "REG"),
+#'   identifier_type = c("code", "code"),
+#'   identifier = c("E92000001", "E12000004")
+#' )
+#'
+#' parse_tojson_params(
+#'   example_id("indicator"),
+#'   time_periods = "2024|W23",
+#'   geographies = dfgeographies,
+#'   filter_items = c("pmRSo")
+#' ) |>
+#'   cat()
+#'
+#' # Create a filter list to find the combination of:
+#' #   - day_number is in c("uLQo4", "qf0jG", "aMjLP") *and*
+#' #   - reason is in c("bBrtT", "ThjPJ", "hsHyW", "m2m9K") *and*
+#' #   - education_phase is in c("5UNdi", "crH31")
+#' filter_list <- list(
+#'   day_number = c("uLQo4", "qf0jG", "aMjLP"),
+#'   reason = c("bBrtT", "ThjPJ", "hsHyW", "m2m9K"),
+#'   education_phase = c("5UNdi", "crH31")
+#' )
+#'
+#' parse_tojson_params(
+#'   example_id("indicator"),
+#'   time_periods = "2024|W23",
+#'   geographies = "NAT|code|E92000001",
+#'   filter_items = filter_list
+#' ) |>
+#'   cat()
+#'
+parse_tojson_params <- function(
+    indicators,
+    time_periods = NULL,
+    geographies = NULL,
+    filter_items = NULL,
+    page = 1,
+    page_size = 1000,
+    verbose = FALSE) {
+  # Set some default strings
+  bridge <- "\n  ]\n},"
+  debug_str <- ",\n\"debug\": true"
```

Comment:
Need to turn the debug off

![image](https://github.com/user-attachments/assets/6ba65160-1329-49f2-a75e-3a0a9ff2ac45)

Reply:
Done